test/lit/passes/inlining_optimize-level=3.wast

```diff
@@ -220,5 +230,197 @@
     (drop (call $no-loops-but-one-use-but-exported))
     (drop (call $no-loops-but-one-use-but-tabled))
   )
-)
 
+  ;; Two functions that each call each other. (Exported, so that they are not
+  ;; removed after inlining.) We should only perform a limited amount of
+  ;; inlining here - a little might help, but like loop unrolling it loses its
+  ;; benefit quickly. Specifically here we will see the infinite recursion after
+  ;; one inlining, and stop.
```

Comment:
>  Specifically here we will see the infinite recursion after one inlining, and stop.

How was this possible? We haven't exceeded the number of functions within this module, and this function hasn't been inlined into 5 times yet..

Reply:
We have a specific check to avoid inlining a function into itself. So once we see that, we stop.

(Infinite recursion can still happen but it must be mutual recursion of multiple functions together.)